packages/cxl-ui/src/components/cxl-credential.js

```diff
@@ -0,0 +1,191 @@
+
+  @property({ type: Number }) scale = 0.5;
+
+  @property({ type: Number, attribute: 'max-width' }) maxWidth;
+
+  @property({ type: Number, attribute: 'parent-padding' }) parentPadding = 0;
+
+  @property({ type: Number, attribute: 'code-scale' }) codeScale = 8;
+
+  @query('#qr-code') qrCodeCanvas = null;
+
+  defaultUrl ='https://cxl.com/institute/credential'
+  
+  get url() {
+    return `${this.credentialUrl || this.defaultUrl}/${this.credentialId}`;
+  }
+
+  get formattedDate() {
+    const date = new Date(this.date);
+    const options = { year: 'numeric', month: 'long', day: 'numeric' };
+    return date.toLocaleDateString('en-US', options);
+  }
+
+  /**
+   *
+   * @param {number} stroke
+   * @param {number} size
+   * @returns {TemplateResult}
+   */
+  static _circle(stroke, size) {
+    return html`
+      <svg
+        width=${size}
+        height=${size}
+        viewBox="0 0 ${size + stroke} ${size + stroke}"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <circle
+          cx="${(size + stroke) / 2}"
+          cy="${(size + stroke) / 2}"
+          r="${size / 2}"
+          fill="none"
+          stroke-width="${stroke}"
+          stroke="rgba(255,255,255,0.15)"
+        />
+      </svg>
+    `;
+  }
+
+  /**
+   * Generate background concentric circles decoration
+   * @param {number} numberOfCircles
+   * @returns {TemplateResult}
+   */
+  static _circleBackground(numberOfCircles) {
+    // Initialize array of n circle diameters,
+    const defaultStroke = 5.5;
+    const radiusIncrease = 50;
+    const arr = Array(numberOfCircles).fill(defaultStroke);
+    return html` ${arr.map((stroke, i) => CXLCredentialElement._circle(stroke, stroke + i * radiusIncrease))} `;
+  }
+
+  /**
+   * This function is called when this.scale is defined or changed. In order to
+   * prevent bad rendering on mobile devices, and because of how we use
+   * `transform: scale` to handle resizing, it calculates and applies a
+   * correction multiplier in case the viewport is smaller than the intended
+   * size of the certificate rendering.
+   *
+   * It works well, but is not meant to react to resizing a window or changing
+   * device orientation. But it will adjust, if the page is reloaded after.
+   */
+  _handleScaling() {
+    const width = this.maxWidth || window.innerWidth - this.parentPadding;
+    const finalW = this._certWidth * this.scale;
+    this.correctionFactor = width < finalW || this.maxWidth ? width / finalW : 1;
+    this.style.height = `${this._certHeight * this.scale * this.correctionFactor}px`;
+    this.style.width = `${this._certWidth * this.scale * this.correctionFactor}px`;
+  }
+
+  /**
+   * Render the QR code target the correct html element in the template
+   */
+  _renderQrCode() {
+    QRCode.toCanvas(this.qrCodeCanvas, this.url, {
+      margin: 0,
+      scale: this.codeScale || 8,
+      color: {
+        dark: '#ffffffff',
+        light: '#ffffff00',
```

Comment:
This has too many chars

Reply:
That's using the hex version of alpha channel, necessary to make the background of the qr-code transparent.  Using the css `rgba` function is not supported by the QRCode lib.